### DISCUSSION TITLE
Rename `with_defaults` to `tag_defaults` (or something else)

### DISCUSSION THREAD

**Discussion by paulcsmith** [luckyframework/lucky#1244] [General] [answered]
There is a lot of confusion around `with_defaults`. Part of it is that it is not found on the guides :D But it is also that the name is too generic and a bit confusing.

I think `tag_defaults` would clarify what it is used for and what it does:

```crystal
tag_defaults field: field, class: "input" do |tag_builder|
  yield tag_builder
end

m Shared::Field, &.text_input
```

I'd also like to come up with a good name for the yielded object. People can call it whatever they want, but having a good name for the docs and in auto-generated code like `Shared::Field` would be good

Maybe:

* `defaults`
* `tag_builder`

**Comment by stephendolan** [comment]
We'll want to modify this issue whenever we come up with a solution: https://github.com/luckyframework/website/issues/410

**Reply by paulcsmith** [reply]
Ah yes! Good catch

**Comment by stephendolan** [comment]
@paulcsmith I think that lacking any other strong opinions, we should move this to `tag_defaults` strictly from a searchability perspective. I actually found `with_defaults` just as intuitive to use within an actual component, and didn't feel like the `tag_defaults` renamed really explained the concept any more succinctly.

This is probably one of those things where lots of good screencasts/website docs that are easily accessible will be much more effective for Lucky users than toiling over getting the name "perfect"!

Just to add some historical context, here's how a real-world working use case looks and feels:
**Definition**: https://github.com/stephendolan/lucky_jumpstart/blob/master/src/components/ui/button.cr
**Use**: https://github.com/stephendolan/lucky_jumpstart/blob/master/src/components/shared/navbar.cr

**Reply by paulcsmith** [reply]
Those are super cool examples!

**Comment by paulcsmith** [comment, marked as answer]
Let's go with `tag_defaults`! https://github.com/luckyframework/lucky/issues/1252
